api/resources_portal/views/attachment.py

```diff
@@ -25,28 +25,35 @@ class Meta:
         read_only_fields = ("id", "created_at", "updated_at")
 
 
+def user_has_perm_on_active_material_request(user, perm):
+    for organization in user.organizations.all():
+        if user.has_perm(perm, organization):
+            for material in organization.materials.all():
+                for request in material.requests.all():
+                    if request.is_active:
+                        return True
```

Comment:
I'm concerned this could become a lot of queries at some point. They say premature optimization is the devil, so I'm not sure we should rewrite all of this to be a single query just yet. However perhaps we could use this https://django-guardian.readthedocs.io/en/stable/userguide/performance.html#prefetching-permissions to at least prevent a separate query for each org.

Additionally we should probably prefetch requests, since we know we want to loop over all of them. (I'll make a suggestion for that.)